### ISSUE TITLE
Request hangs when attempting connection with invalid credentials

### ISSUE THREAD

**Issue by kgreenpea** [codemix/oriento#238]
Trying to open a user connection with invalid username and/or password leaves the request hanging, rather than erroring. In the following example, neither error nor success functions are executed if the supplied credentials are invalid.  

	server.use({
		name: 'testdb',
		username: 'invalid_user',
		password: 'invalid_pw'
	})
		.open()
		.then(function (item) {
			// Called successfully with valid credentials
		})
		.catch(function (err) {
			// Not called when invalid
		});

**Comment by phpnode** [comment]
@kgreenpea I added a [test for this](https://github.com/codemix/oriento/blob/master/test/bugs/238-hang-on-invalid-credentials.js) but couldn't replicate the issue, if you can make that test fail I'd be happy to reopen this, thanks!

**Comment by kgreenpea** [comment]
@phpnode Great, thanks for the speedy turnaround with issues! 

I should have mentioned I'm using tokens... So in your test, if you set useToken to true on serverValid, attempting to open a user connection with invalid database credentials against a valid server causes the issue (in my environment). 

With invalid credentials specified for the server, as in your second case, or with useToken set to false the error function is called as expected. 

So for now I'm ok to specify user credentials in a server config rather than on the call to use(), I'll leave it to you to decide if you want to reopen the issue though. Thanks again!

**Comment by phpnode** [comment]
@kgreenpea I've changed the test to use tokens but it still passes, can you confirm this test still fails for you? I can't really help much further with this issue until you can make the test itself fail (PR welcome!)

**Comment by kgreenpea** [comment]
@phpnode I was able to reproduce the issue using your test... So the difference in my case was that I had reused the original global test server instance as the validServer, rather than creating a fresh server following the call to createTestDb, i.e. serverValid = global.TEST_SERVER. 
With the global server set to use tokens the test hangs as described. Without using tokens the test also fails, but differently.

I've modified the test so that the server instance is created and reused locally to demonstrate this, see PR. Uncomment line 83 to make the test pass.

So the lesson (for me) is that the initial server instance used to drop/create a db should not be used again to connect a user. I can live with that so no real urgency, but don't really understand why it should be - perhaps that needs some clarification in the usage documentation if not fixed otherwise?

Thanks again for your attention much appreciated.
